doc_source/constructs.md

```diff
@@ -1,104 +1,100 @@
 
-```
-new SomeConstruct(parent, name[, props]);
+For example, here's how you would define an SNS Topic in your stack with default configuration:
+
+```ts
+new sns.Topic(this, 'MyTopic');
 ```
 
-In almost all cases, you should pass the keyword `this` for the `parent` argument, because you will generally initialize new constructs in the context of the parent construct\. Any descriptive string will do for the `name` argument, and an in\-line object for the set of properties\.
+The first argument to every construct is always the scope in which it is created, and will almost always be simply `this` because most constructs will be defined within the current scope.
 
-```
+Scopes enable constructs to be composed together to form higher-level abstractions by enabling the framework to group them together into logical units, allocate globally unique identifiers and allow them to consult __context information__ such as the region in which it is going to be deployed into, which AZs are available for my account, etc.
+
+In most cases, the construct initializer will have a third `props` argument that can be used to define the construct's initial configuration. For example:
+
+```ts
 new BeautifulConstruct(this, 'Foo', {
-    applicationName: 'myApp',
-    timeout: 300
+  favoriteColor: 'green',
+  timeout: 300
 });
 ```
 
-**Note**  
-Associating the construct to its parent as part of initialization is necessary because the construct occasionally needs contextual information from its parent, such as to which the region the stack is deployed\.
+The `construct.node` property can be used to interact with the construct's node representation:
 
-Use the following operations to inspect the construct tree\.
+* `construct.node.scope` The scope in which this construct was defined.
 
- aws\-cdk\.Construct\.parent   
-Gets the path of this construct from the root of the tree\.
+* `construct.node.id` Returns the "id" of the construct passed upon creation.
 
- aws\-cdk\.Construct\.getChildren   
-Gets an array of all of the contruct's children\.
+* `construct.node.uniqueId` Returns an app-wide unique safe ID of this construct. This ID encodes the construct's path into a human readable portion as well as a hash of the full path to ensure global uniqueness.
 
- aws\-cdk\.Construct\.getChild   
-Gets the child construct with the specified ID\.
+* `construct.node.path` Gets the path of this construct from the root of scope (the `App`)\.
 
- aws\-cdk\.Construct\.toTreeString\(\)   
-Gets a string representing the construct's tree\.
+## Construct IDs<a name="constructs_ids"></a>
 
-## Construct Names<a name="constructs_EVER"></a>
+Every construct in a CDK app must have an **id** unique within the scope in which the construct is defined. IDs are used to find constructs in the construct hierarchy, and to allocate logical IDs so that AWS CloudFormation can keep track of the generated resources\.
```

Comment:
First mention of the hierarchy is here.

Reply:
Scope and hierarchy are fundamental concepts.

Reply:
They still need to be explained in the context of the CDK in my mind.